### PR TITLE
Add a global cache layer common for all the endpoints

### PR DESCRIPTION
**Pull request by taik0**
to reduce pressure on the JWK backends.

### CODE REVIEW

File: key_cacher.go [hunk 1]
```diff
@@ -85,6 +161,21 @@ func NewMemoryKeyCacher(maxKeyAge time.Duration, maxCacheSize int, keyIdentifySt
 	}
 }
 
+func NewGlobalMemoryKeyCacher(maxKeyAge time.Duration, maxCacheSize int, keyIdentifyStrategy string) KeyCacher {
+	if keyIdentifyStrategy == "" {
+		keyIdentifyStrategy = "kid"
+	}
```
Comment (dhontecillas) [inline key_cacher.go:167]: here is where I would check if we already have a reference:
```go
var gc GlobalCacher
if kc, ok := globalKeyCacher[keyIdentifyStrategy]; ok {
    gc := kc
} else {
    return nil, fmt.Errorf("invalid strategy %s", gkc.strategy)
}
```

File: key_cacher.go [hunk 2]
```diff
@@ -3,19 +3,61 @@ package jose
...
+type GlobalCacher struct {
+	kc KeyCacher
+	mu *sync.RWMutex
+}
+
+func SetGlobalCacher(l logging.Logger, cfg config.ExtraConfig) error {
+	duration, err := configGetter(cfg)
+	if err != nil {
+		if err != ErrNoValidatorCfg {
+			l.Error("[SERVICE: JWTValidator]", err.Error())
```
Comment (alombarte) [inline key_cacher.go:35]: `[SERVICE: JOSE]`

File: key_cacher.go [hunk 3]
```diff
@@ -3,19 +3,61 @@ package jose
...
+	})
+	return nil
+}
+
+func configGetter(cfg config.ExtraConfig) (time.Duration, error) {
+	e, ok := cfg[ValidatorNamespace].(map[string]interface{})
+	if !ok {
+		return defaultGlobalCacheMaxAge, fmt.Errorf("no config")
+	}
+	duration, ok := e["cache_duration"].(string)
```
Comment (alombarte) [inline key_cacher.go:54]: `shared_cache_duration`

File: key_cacher.go [hunk 4]
```diff
@@ -3,19 +3,61 @@ package jose
...
+	if !ok {
+		return defaultGlobalCacheMaxAge, fmt.Errorf("no duration")
+	}
+	return time.ParseDuration(duration)
```
Comment (alombarte) [inline key_cacher.go:58]: Duration should be an uint32 expressing seconds, as we have on the `cache_duration`

**Review by alombarte — CHANGES_REQUESTED** [review]
The namespace `auth/jose` is needed to load this configuration at the service level

**Review by kpacha — APPROVED** [review]

**Review by alombarte — CHANGES_REQUESTED** [review]

**Pull request merged**